utils/src/test/scala/za/co/absa/enceladus/utils/transformations/DeepArrayErrorTransformationSuite.scala

```diff
@@ -857,6 +857,97 @@ class DeepArrayErrorTransformationSuite extends FunSuite with SparkTestBase with
+
+    val df = JsonUtils.getDataFrameFromJson(spark, Seq(sample))
+
+    val expectedSchema =
+      """root
+        | |-- id: long (nullable = true)
+        | |-- legs: array (nullable = true)
+        | |    |-- element: struct (containsNull = false)
+        | |    |    |-- conditions: array (nullable = true)
+        | |    |    |    |-- element: struct (containsNull = false)
+        | |    |    |    |    |-- amount: long (nullable = true)
+        | |    |    |    |    |-- checks: array (nullable = true)
+        | |    |    |    |    |    |-- element: struct (containsNull = false)
+        | |    |    |    |    |    |    |-- checkNums: array (nullable = true)
+        | |    |    |    |    |    |    |    |-- element: string (containsNull = true)
+        | |    |    |    |    |    |    |-- optimizedNums: array (nullable = true)
+        | |    |    |    |    |    |    |    |-- element: string (containsNull = true)
+        | |    |    |-- legid: long (nullable = true)
+        | |-- errors: array (nullable = true)
+        | |    |-- element: struct (containsNull = true)
+        | |    |    |-- errType: string (nullable = true)
+        | |    |    |-- errCode: string (nullable = true)
+        | |    |    |-- errMsg: string (nullable = true)
+        | |    |    |-- errCol: string (nullable = true)
+        | |    |    |-- rawValues: array (nullable = true)
+        | |    |    |    |-- element: string (containsNull = true)
+        | |    |    |-- mappings: array (nullable = true)
+        | |    |    |    |-- element: struct (containsNull = true)
+        | |    |    |    |    |-- mappingTableColumn: string (nullable = true)
+        | |    |    |    |    |-- mappedDatasetColumn: string (nullable = true)
+        |""".stripMargin.replace("\r\n", "\n")
+
+    val expectedResults =
+      """[ {
+        |  "id" : 1,
+        |  "legs" : [ {
+        |    "conditions" : [ {
+        |      "amount" : 100,
+        |      "checks" : [ {
+        |        "checkNums" : [ "1", "2", "3b", "4", "5c", "6" ],
+        |        "optimizedNums" : [ "1_100_1", "2_100_1", "3b_100_1", "4_100_1", "5c_100_1", "6_100_1" ]
+        |      } ]
+        |    } ],
+        |    "legid" : 100
+        |  } ],
+        |  "errors" : [ {
+        |    "errType" : "confCastError",
+        |    "errCode" : "E00003",
+        |    "errMsg" : "Conformance Error - Null returned by casting conformance rule",
+        |    "errCol" : "legs.conditions.checks.optimizedNums",
+        |    "rawValues" : [ "3b" ],
+        |    "mappings" : [ ]
+        |  }, {
+        |    "errType" : "confCastError",
+        |    "errCode" : "E00003",
+        |    "errMsg" : "Conformance Error - Null returned by casting conformance rule",
+        |    "errCol" : "legs.conditions.checks.optimizedNums",
+        |    "rawValues" : [ "5c" ],
+        |    "mappings" : [ ]
+        |  } ]
+        |} ]"""
+        .stripMargin.replace("\r\n", "\n")
```

Comment:
I wanted to be per-file consistent. Since all other tests in this file are written this way, a new test follows the pattern.

My plan is to move this to 'spark-hats' permanently (code and tests), and then make everything consistent inside 'spark-hats'.